test/boost/continuous_data_consumer_test.cc

```diff
@@ -114,3 +114,46 @@ SEASTAR_THREAD_TEST_CASE(test_read_unsigned_vint) {
     }
 }
 
+class test_consumer2 final : public data_consumer::continuous_data_consumer<test_consumer2> {
+    bool _reading = false;
+
+public:
+    test_consumer2() : continuous_data_consumer(tests::make_permit(), make_buffer_input_stream(
+                                       temporary_buffer<char>("abababababababababab", 20),
+                                       [] {return 1;}), 0, 10)
+    { }
+
+    bool non_consuming() { return false; }
```

Comment:
I don't understand why this test overrides the consumer's methods instead of just using them.
I'm worried that if a year from now somebody changes the implementation of the consumer, this test will stop working, and nobody will understand how to fix the test because it's not clear what it is actually trying to test.
At the very minimum we need a comment of what this test is trying to test, but if there's a way to reproduce it without overriding all the methods, it will be even better.

Reply:
We can't actually use the `continuous_data_consumer` without overriding at least `verify_end_state()` and `non_consuming()` because in `continuous_data_consumer` these methods actually use the corresponding overriding methods (using the template type from `: public data_consumer::continuous_data_consumer<test_consumer2>`).
I can remove overriding `fast_forward_to()` though I overrode it to indicate that if the assert passes, the problem comes from the consumer, not from the test itself

Reply:
I've upgraded the test, so @nyh you can take a look if it's good now